service/src/main/java/uk/gov/hmcts/reform/fpl/service/ActionCmoService.java

```diff
@@ -1,26 +1,48 @@
 package uk.gov.hmcts.reform.fpl.service;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
 import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
 import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.HearingDateDynamicElement;
+import uk.gov.hmcts.reform.fpl.model.OrderAction;
 import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+
+import java.time.LocalDateTime;
 
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 
 @Service
 public class ActionCmoService {
+    private final ObjectMapper objectMapper;
     private final DraftCMOService draftCMOService;
+    private final DateFormatterService dateFormatterService;
+    private final HearingVenueLookUpService hearingVenueLookUpService;
+    private final HearingBookingService hearingBookingService;
 
     private static final String CMO_ACTION_KEY = "orderAction";
     private static final String CMO_KEY = "caseManagementOrder";
 
     //TODO: this should all exist in one CaseManagementOrderService
```

Comment:
is this happening as part of this story, or another story currently in progress, or in future? 😄

Reply:
Branch was based from FPLA-24 which this comment came from (@swalker125 made it). I think Sam plans to make this change as part of FPLA-24 but i'd be keen on pushing this to tech debt just until release is over.

Reply:
tech debt. draftCMO service need huge changes and I am not bring them into 929 or 24